src/ast/passes/semantic_analyser.cpp

```diff
@@ -2803,6 +2811,48 @@ void SemanticAnalyser::visit(AssignConfigVarStatement &assignment)
   Visit(assignment.expr);
 }
 
+void SemanticAnalyser::visit(VarDeclStatement &decl)
+{
+  std::string var_ident = decl.var->ident;
+
+  // This should be the first time we're seeing this variable
+  auto decl_search = declared_variables_[scope_].find(var_ident);
+  if (decl_search != declared_variables_[scope_].end()) {
```

Comment:
C++17 baby! To keep the search variable scoped to where it's needed:
```suggestion
  if (auto decl_search = declared_variables_[scope_].find(var_ident);
      decl_search != declared_variables_[scope_].end()) {
```

(a lot of the old code was written pre-C++17 which is why it doesn't do this)